app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumberForm.kt

```diff
@@ -240,27 +257,38 @@ class AddHousenumberForm : AbstractQuestFormAnswerFragment<HousenumberAnswer>()
         }
     }
 
-    private fun createAnswer(): HousenumberAnswer? =
-        if (houseNameInput != null) {
-            houseNameInput?.nonEmptyInput?.let { HouseName(it) }
-        }
-        else if (conscriptionNumberInput != null && streetNumberInput != null) {
-            conscriptionNumberInput?.nonEmptyInput?.let { conscriptionNumber ->
-                val streetNumber = streetNumberInput?.nonEmptyInput // streetNumber is optional
-                ConscriptionNumber(conscriptionNumber, streetNumber)
-            }
+    override fun isRejectingClose(): Boolean {
+        val houseName = houseNameInput?.nonEmptyInput
+        val houseNumber = houseNumberInput?.nonEmptyInput
+        val conscriptionNumber = conscriptionNumberInput?.nonEmptyInput
+        val streetNumber = streetNumberInput?.nonEmptyInput
+        val blockNumber = blockNumberInput?.nonEmptyInput
+
+        if(houseName != null || houseNumber != null) {
+            return true
         }
-        else if (blockNumberInput != null && houseNumberInput != null) {
-            blockNumberInput?.nonEmptyInput?.let { blockNumber ->
-                houseNumberInput?.nonEmptyInput?.let { houseNumber ->
-                    HouseAndBlockNumber(houseNumber, blockNumber)
-                }
-            }
+        if(conscriptionNumber != null || streetNumber != null || blockNumber != null) {
+            return true
         }
-        else if (houseNumberInput != null) {
-            houseNumberInput?.nonEmptyInput?.let { HouseNumber(it) }
+        return false
```

Comment:
You could replace this with `return houseName != null || houseNumber != null || conscriptionNumber != null || streetNumber != null || blockNumber != null`. No need for separate `if` statements that return a boolean.

Reply:
I know but it seemed better than super long line

Reply:
Conceptually, I find it harder to understand though. It seems that there are three different conditions, when really there is just one.

> super long line

Why not like this then?

```kt
return houseName != null || houseNumber != null || conscriptionNumber != null
    || streetNumber != null || blockNumber != null
```

Reply:
Somehow I prefer multiple ifs over splitting line this way, but maybe it is just me.

Reply:
How about
```kt
return listOf(houseName, houseNumber, conscriptionNumber, streetNumber, blockNumber).any { it != null }
```

Reply:
I agree that @FloEdelmann's version has better semantics.

I suspect that @matkoniecz likes the current version because of the grouping of the values.

You can get both using intermediary values
```kt
val hasNameOrNumber = houseName != null || houseNumber != null
val hasWhateverTheseAreCalled = conscriptionNumber != null || streetNumber != null || blockNumber != null
return hasNameOrNumber || hasWhateverTheseAreCalled
```